include/kernel.h

```diff
@@ -3959,6 +3976,219 @@ static inline char *K_THREAD_STACK_BUFFER(k_thread_stack_t sym)
 
 #endif /* _ARCH_DECLARE_STACK */
 
+/**
+ * @defgroup mem_domain_apis Memory domain APIs
+ * @ingroup kernel_apis
+ * @{
+ */
+
+/** @def MEM_PARTITION_ENTRY
+ *  @brief Used to declare a memory partition entry
+ */
+#define MEM_PARTITION_ENTRY(_start, _size, _attr) \
+	{\
+		.start = _start, \
+		.size = _size, \
+		.attr = _attr, \
+	}
+
+/** @def K_MEM_PARTITION_DEFINE
+ *  @brief Used to declare a memory partition
+ */
+#ifdef _ARCH_MEM_PARTITION_ALIGN_CHECK
+#define K_MEM_PARTITION_DEFINE(name, start, size, attr) \
+	_ARCH_MEM_PARTITION_ALIGN_CHECK(start, size); \
+	struct k_mem_partition name = \
+		MEM_PARTITION_ENTRY((u32_t)start, size, attr)
+#else
+#define K_MEM_PARTITION_DEFINE(name, start, size, attr) \
+	struct k_mem_partition name = \
+		MEM_PARTITION_ENTRY((u32_t)start, size, attr)
+#endif /* _ARCH_MEM_PARTITION_ALIGN_CHECK */
+
+/* memory partition arch/soc independent attribute */
+/** @def K_MEM_PARTITION_P_NA_U_NA
+ *  @brief Denote partition is "Privileged No Access, Unprivileged No Access"
+ */
+#define K_MEM_PARTITION_P_NA_U_NA				0
```

Comment:
That would be  'K_MEM_PARTITION_P_NA_U_NA' and 'K_MEM_PARTITION_P_RW_U_RO' we cant set such properties in the MMU page tables.

'K_MEM_PARTITION_P_NA_U_NA' doesn't make sense to me because in the privilege mode can access anything it wants to.

'K_MEM_PARTITION_P_RW_U_RO' I can't specify different access rights to privileged and unprivileged threads. So I can only set both privileged and unprivileged to either RO or RW. its something like 'K_MEM_PARTITION_P_RW_U_RW' or 'K_MEM_PARTITION_P_RO_U_RO'